Chessington.GameEngine/Pieces/Piece.cs

```diff
@@ -6,12 +6,108 @@ namespace Chessington.GameEngine.Pieces
 {
     public abstract class Piece
     {
+        protected Square currentSquare;
+        protected Board board;
+
+        public Player Player { get; private set; }
+
         protected Piece(Player player)
         {
             Player = player;
         }
 
-        public Player Player { get; private set; }
+        protected bool OpponentSquare(Square square)
+        {
+            var piece = board?.GetPiece(square);
+            if (piece == null)
+            {
+                return false;
+            }
+
+            return board.GetPiece(square).Player != Player;
+        }
+
+        protected bool FriendlySquare(Square square)
+        {
+            var piece = board?.GetPiece(square);
+            if (piece == null)
+            {
+                return false;
+            }
+
+            return board.GetPiece(square).Player == Player;
+        }
+
+        private Square GetNextSquare(int distance, int rowDirection, int colDirection)
+        {
+            var nextRow = currentSquare.Row + distance * rowDirection;
+            var nextCol = currentSquare.Col + distance * colDirection;
+            return Square.At(nextRow, nextCol);
+        }
+
+        private IEnumerable<Square> MovesOnDirection(int rowDirection, int colDirection)
+        {
+            var moves = new List<Square>();
+
+            for (var distance = 1; distance < 7; distance++)
```

Comment:
Instead of using explicit numbers, you can access the size of the board from `GameSettings.BoardSize`. eg. `GameSettings.BoardSize - 1 == 7`

Reply:
Good point